### PR TITLE
[TT-274] Close target connection when request is canceled

### PR DESCRIPTION
**Pull request by gernest**
backport changes added to the net/http/httputil ReverseProxy for go1.15 .
See release notes here https://golang.org/doc/go1.15#net/http/httputil

## Description
backport changes added to the net/http/httputil ReverseProxy for go1.15 .
See release notes here https://golang.org/doc/go1.15#net/http/httputil

## Related Issue
https://github.com/golang/go/issues/35559

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

**Comment by gernest** [comment]
Most of the files are a result of `go mod vendor` The changes themselves are very minimal.

**Comment by gernest** [comment]
rebased, all the vendor files are gone now

**Review by sredxny — APPROVED** [review]
@gernest looks good to me. Is there any Jira or GH issue that we can reference to this fix?

**Comment by gernest** [comment]
@sredxny we are using a modified copy of `httputil.ReverseProxy` from go std distribution. An actual issue is from go project  https://github.com/golang/go/issues/35559. So we are just backporting changes here to keep our ReverseProxy up to date with the upstream .

File: vendor/modules.txt [hunk 1]
```diff
@@ -150,7 +150,7 @@ github.com/imdario/mergo
 github.com/jensneuse/abstractlogger
 # github.com/jensneuse/byte-template v0.0.0-20200214152254-4f3cf06e5c68
 github.com/jensneuse/byte-template
-# github.com/jensneuse/graphql-go-tools v0.0.0-00010101000000-000000000000 => github.com/TykTechnologies/graphql-go-tools v1.6.2-0.20200918115423-1f1792f5b6d9
+# github.com/jensneuse/graphql-go-tools v0.0.0-00010101000000-000000000000 => github.com/TykTechnologies/graphql-go-tools v1.6.2-0.20200731074614-80c67fc17e8e
```
Comment (spetrunin) [inline vendor/modules.txt:153]: @gernest could you revert this? it is an incorrect version
and go 1.14 will check consistency between go.mod and modules.txt

Comment (gernest) [reply vendor/modules.txt]: @spetrunin fixed  9cb38999b3277661f063dc16846347b6f570ef9a

**Review by matiasinsaurralde — APPROVED** [review]

**Comment by matiasinsaurralde** [comment]
@gernest LGTM, there are some conflicts on the tests.

**Comment by furkansenharputlu** [comment]
Merged manually to `release-4`

**Pull request merged**